pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -69,6 +69,10 @@ type RolloutSpec struct {
 	// Note that progress will not be estimated during the time a rollout is paused.
 	// Defaults to 600s.
 	ProgressDeadlineSeconds *int32 `json:"progressDeadlineSeconds,omitempty" protobuf:"varint,8,opt,name=progressDeadlineSeconds"`
+	// AbortExceedProgressDeadline to whether abort the update when ProgressDeadlineSeconds
+	// is exceeded if analysis is not used. Default is false.
+	// +optional
+	AbortExceedProgressDeadline *bool `json:"abortExceedProgressDeadline,omitempty" protobuf:"varint,12,opt,name=abortExceedProgressDeadline"`
```

Comment:
Since:
1. the default of this feature is false
2. false booleans will be omitempty
3. and we don't need to distinguish unspecified vs. false

I think it would be better to change this to be a `bool` instead of a `*bool`

Reply:
I propose we rename this to `progressDeadlineAbort` so that field names are prefixed consistently. i.e.

```yaml
spec:
    progressDeadlineSeconds: 600
    progressDeadlineAbort: true
```

Reply:
Thanks @jessesuen , will update the PR.

Reply:
Done